server/src/main/scala/org/http4s/server/middleware/VirtualHost.scala

```diff
@@ -69,11 +72,14 @@ object VirtualHost {
           val host: Host = h.port match {
             case Some(_) => h
             case None =>
-              h.copy(port = req.uri.port.orElse(req.isSecure.map(if (_) Uri.Port.https else Uri.Port.http)))
+              h.copy(
+                port =
+                  req.uri.port.orElse(req.isSecure.map(if (_) Uri.Port.https else Uri.Port.http)))
           }
           (first +: rest).toVector
             .collectFirst { case HostService(s, p) if p(host) => s(req) }
-            .getOrElse(OptionT.liftF(Response[F](NotFound).withBody(s"Host '$host' not found.")))
+            .getOrElse(
+              OptionT.liftF(Response[F](NotFound).withBody(s"Host '${host.value}' not found.")))
```

Comment:
This echoes request information to the response, which we don't do because of reflection attack paranoia.  I'm not sure why this doesn't return None anyway.